File: examples/hello.ts
// Hello World in ScriptRust

/* xxx, immutable: greeting message */
const message: string = "Hello, ScriptRust!";

/* xxx, pure: simple greeting function */
function greet(/* xxx, immutable: name */ name: string): string {
  return "Hello, " + name + "!";
}

console.log(message);
console.log(greet("World"));
console.log(greet("Developer"));

File: examples/hello.rs
fn main() {
    let message: String = "Hello, ScriptRust!";

    fn greet(name: String) -> String {
        format!("Hello, {}!", name)
}

    println!("{:?}", message);

    println!("{:?}", greet("World"));

    println!("{:?}", greet("Developer"));

}
